### ISSUE TITLE
[Loss function used]

### ISSUE THREAD

**Issue by baibizhe** [serdarozsoy/corinfomax-ssl#3]
Hello .Thanks for your wonderful work.
I saw you have use CovarianceLossv2() in  [imagenet1k](https://github.com/serdarozsoy/corinfomax-ssl/tree/main/imagenet1k)/main_orig_tinyimg_clone_mgpu_corrected_offline_bn.py    . Is this the correct loss function ?Because I saw there is a LDMILoss() in [imagenet1k](https://github.com/serdarozsoy/corinfomax-ssl/tree/main/imagenet1k)/loss.py . In addition . what's the difference between those loss function  in loss.py ?  Thanks.

**Comment by serdarozsoy** [comment]
Thank you for your interest. I can confirm that `CovarianceLossv2()` in [imagenet1k/main_orig_tinyimg_clone_mgpu_corrected_offline_bn.py](https://github.com/serdarozsoy/corinfomax-ssl/blob/main/imagenet1k/main_orig_tinyimg_clone_mgpu_corrected_offline_bn.py) is the correct loss function as presented in the code. 

To maintain reproducibility by avoiding unintentional changes to the code, we did not make any significant edits when submitting it to NeurIPS and sharing it on Github. Within the original code, we included some of the loss functions that we explored during the development process.

`LDMILoss()` is one of them. It directly uses LDMI loss as in Equation (3) in the paper, rather than the latest form of the loss, CorInfoMax. CorInfoMax is a first-order approximation of LDMI and it constrains the linear dependence to the identity mapping. For more information on the evolution of LDMI to CorInfoMax, you can refer to Section 3 of our paper.

**Comment by baibizhe** [comment]
Thanks!
When I tried to reproduce without any change based on the cifar10 code, the nan training loss occurred in the pretraining procedure(Happened in the second epoch). Are there any suggestions for hyper-parameter tuning for cifar10?

Best regards,
Bizhe

**Comment by serdarozsoy** [comment]
If you are using the datasets from the paper, there is no need to do hyper-parameter tuning again. If you do not modify the code, and you use the run command code we shared, it is not possible to get a "nan" value when pretraining. Please make sure that you copy and use the run command in README.md correctly. 

For testing this quickly, you can also download the repository and obtain the results with the shared run command, as I did on Google Colab: [https://colab.research.google.com/drive/1klszlh2Cyg9RBioc2MA9o7hvAlPUEepR?usp=sharing](https://colab.research.google.com/drive/1klszlh2Cyg9RBioc2MA9o7hvAlPUEepR?usp=sharing
)

I am also reposting the shared run command for CIFAR-10 in the main README.md (Part: “Commands for original runs”) : 

> python main_traintest.py --epochs 1000 --batch_size 512 --lin_epochs 100 --lin_batch_size 256 --R_ini 1.0  --learning_rate 0.5 --cov_loss_weight 1.0 --sim_loss_weight 250.0 --la_R 0.01 --la_mu 0.01 --projector 2048-2048-64 --R_eps_weight 1e-8 --w_decay 1e-4 --pre_optimizer SGD --pre_scheduler lin_warmup_cos --lin_optimizer SGD --lin_learning_rate 0.2 --lin_w_decay 0 --lin_scheduler cos  --n_workers 4  --dataset cifar10 --lin_dataset cifar10 --con_name cov_cifar10_best_rerun --model_name resnet18 --normalize_on --min_lr 1e-6  --lin_min_lr 0.002

**Comment by baibizhe** [comment]
Thank a lot for your detailed explanation.
I try your command on imagenet-tiny with command provided in readme "python ./main_traintest.py --epochs 800 --batch_size 1024 --lin_epochs 100 --lin_batch_size 256 --R_ini 1.0  --learning_rate 0.5 --cov_loss_weight 1.0 --sim_loss_weight 500.0 --la_R 0.1 --la_mu 0.1 --projector 4096-4096-128 --R_eps_weight 1e-8 --w_decay 1e-4 --warmup_epochs 10 --pre_optimizer SGD --pre_scheduler lin_warmup_cos --lin_optimizer SGD --lin_learning_rate 0.2 --lin_w_decay 0 --lin_scheduler cos  --n_workers 4  --dataset tiny_imagenet --lin_dataset tiny_imagenet --con_name cov_tiny --model_name resnet50 --normalize_on --min_lr 1e-6 --lin_min_lr 0.002"

Here is what I got. It seems that model is overfitting on the classification.
![1677915169041](https://user-images.githubusercontent.com/33809209/222882535-01076b33-e6f6-46e1-86cc-f2c50c1854fb.png)


[t_1677869856_cov_tiny.pth.zip](https://github.com/serdarozsoy/corinfomax-ssl/files/10887742/t_1677869856_cov_tiny.pth.zip)

**Comment by serdarozsoy** [comment]
This issue is not related to overfitting. It is due to the difference in the folder structure of tiny-imagenet's training and test dataset. The dataset to be used for testing should be configured like the training dataset. Please follow the instruction on README of [cifars-tiny](https://github.com/serdarozsoy/corinfomax-ssl/tree/main/cifars-tiny)  folder (probably you missed to run the `python rearrange_folders_tiny.py`):

>Tiny ImageNet is downloaded with 'bash download_tiny.sh', then folders are organize with 'python rearrange_folders_tiny.py'.

**Comment by serdarozsoy** [comment]
This issue was closed as it has not been active for 2 months since the last reply.
